packages/shared/components/inputs/Pin.svelte

```diff
@@ -87,18 +84,41 @@
     }
 
     /**
-     * for android mobile we need both onkeydown and oninput
-     * event listeners to the input and handle the old and the new value.
-     * the auto-suggest feature or other event might follow
-     * the keydown event and invalidate it.
+     * On Android we need both on:keydown and on:input.
+     * Keydown only handles 'Backspace' since some soft-keyboards
+     * doesn't send the inputType value as 'deleteContentBackward'.
+     * Input event handle the rest, as input also could be dictated, drawed, etc.
      */
-    const changeHandlerHelper = (e, i: number) => {
-        if (!/^[0-9]$/.test(e.data)) {
+    function changeHandlerMobile(event: Event & InputEventInit, i: number): void {
+        if (!event.isTrusted || !/^[0-9]$/.test(event.data)) {
             inputs[i] = ''
-        } else {
-            inputElements[i + 1].focus()
+            inputElements[i].focus()
+            return
+        }
+        if (event.inputType === 'deleteContentBackward') {
+            handleBackspace()
+        } else if (event.inputType === 'insertText') {
+            inputElements[i + 1]?.focus()
+            value = inputs.join('')
+            if (validatePinFormat(value)) {
+                dispatch('submit')
+            }
         }
     }
+
+    function handleBackspaceMobile(event: KeyboardEvent): void {
+        if (event.key === 'Backspace') {
+            handleBackspace()
+        }
+    }
+
+    function handleNextMobile(i: number): void {
+        if (inputElements[i - 1].value === '') {
```

Comment:
Im getting a type error here trigger right after the autofocus
```bash
TypeError: Cannot read properties of undefined (reading 'value')
```
Adding optional chaining seems to fix the issue

```suggestion
        if (inputElements[i - 1]?.value === '') {
```